automation/patch_igz/patch_remote.py

```
@@ -295,29 +289,36 @@ def _make_targets(
                 # Set the MLRUN_KFP_IMAGE environment variable in the mlrun-api deployment patch,
                 # so that workflow pods will use the correct KFP image from the internal registry.
                 _, overwrite_registry = self._resolve_overwrite_registry()
-                kfp_image_uri = (
-                    f"{overwrite_registry}/mlrun/{Constants.mlrun_kfp}:{image_tag}"
-                )
-
-                mlrun_api_container = self._deploy_patch["mlrun_api"]["spec"][
-                    "template"
-                ]["spec"]["containers"][0]
-                env_vars = mlrun_api_container.setdefault("env", [])
-                existing_var = next(
-                    (var for var in env_vars if var.get("name") == "MLRUN_KFP_IMAGE"),
-                    None,
+                docker_registry = self._resolve_docker_registry()
+                kfp_registry = (
+                    overwrite_registry if overwrite_registry else docker_registry
                 )
-
-                if existing_var:
-                    existing_var["value"] = kfp_image_uri
-                else:
-                    env_vars.append(
-                        {
-                            "name": "MLRUN_KFP_IMAGE",
-                            "value": kfp_image_uri,
-                        }
+                if kfp_registry:
```

If both `OVERWRITE_IMAGE_REGISTRY` and `DOCKER_REGISTRY` are missing, `MLRUN_KFP_IMAGE` is not set. Is this a valid scenario, or should we fail fast or at least log a warning?

---

you will fail much earlier